src/services/codefixes/interfaceFixes.ts

```diff
@@ -0,0 +1,268 @@
+
+            if (textChanges.length > 0) {
+                return [{
+                    description: getLocaleSpecificMessage(Diagnostics.Implement_interface_on_class),
+                    changes: [{
+                        fileName: sourceFile.fileName,
+                        textChanges: textChanges
+                    }]
+                }];
+            }
+
+            return undefined;
+        }
+    });
+
+    registerCodeFix({
+        errorCodes: [Diagnostics.Non_abstract_class_0_does_not_implement_inherited_abstract_member_1_from_class_2.code],
+        getCodeActions: (context: CodeFixContext) => {
+            const sourceFile = context.sourceFile;
+            const start = context.span.start;
+            const token = getTokenAtPosition(sourceFile, start);
+            const checker = context.program.getTypeChecker();
+
+            let textChanges: TextChange[] = [];
+
+            if (token.kind === SyntaxKind.Identifier && token.parent.kind === SyntaxKind.ClassDeclaration) {
+                const classDeclaration = <ClassDeclaration>token.parent;
+                const startPos = classDeclaration.members.pos;
+                const classMembers = getClassMembers(classDeclaration);
+                const trackingAddedMembers: string[] = [];
+                const extendsClause = ts.getClassExtendsHeritageClauseElement(classDeclaration);
+                textChanges = textChanges.concat(getChanges(extendsClause, classMembers, startPos, checker, /*reference*/ false, trackingAddedMembers, context.newLineCharacter));
+            }
+
+            if (textChanges.length > 0) {
+                return [{
+                    description: getLocaleSpecificMessage(Diagnostics.Implement_inherited_abstract_class),
+                    changes: [{
+                        fileName: sourceFile.fileName,
+                        textChanges: textChanges
+                    }]
+                }];
+            }
+
+            return undefined;
+        }
+    });
+
+    function getChanges(interfaceClause: Node, existingMembers: string[], startPos: number, checker: TypeChecker, reference: boolean, trackingAddedMembers: string[], newLineCharacter: string): TextChange[] {
+        const type = checker.getTypeAtLocation(interfaceClause);
+        const changesArray: TextChange[] = [];
+
+        if (type && type.symbol && type.symbol.declarations) {
+            const interfaceMembers = getMembers(<InterfaceDeclaration>type.symbol.declarations[0], checker);
+            for (let j = 0; interfaceMembers && j < interfaceMembers.length; j++) {
+                if (interfaceMembers[j].name && existingMembers.indexOf(interfaceMembers[j].name.getText()) === -1) {
+                    if (interfaceMembers[j].kind === SyntaxKind.PropertySignature) {
+                        const interfaceProperty = <PropertySignature>interfaceMembers[j];
+                        if (trackingAddedMembers.indexOf(interfaceProperty.name.getText()) === -1) {

```

Comment:
this does not handle computed properties correctly. e.g. `[Symbol.Iterator]`. please add tests.